 Repository.mk                                                |    2 +-
 bean/com/sun/star/comp/beans/LocalOfficeConnection.java      |    2 +-
 bean/native/unix/com_sun_star_beans_LocalOfficeWindow.c      |   10 ++++++----
 bean/native/unix/com_sun_star_comp_beans_LocalOfficeWindow.c |    7 +++++--
 scp2/source/ooo/file_library_ooo.scp                         |   11 ++++++++++-
 5 files changed, 23 insertions(+), 9 deletions(-)

New commits:
commit 708229dc96533b6f165b82ce4d9e07537d1ce8b1
Author: Stephan Bergmann <sbergman@redhat.com>
Date:   Mon May 7 16:54:43 2012 +0200

    fdo#49580: Fix bean after gbuild'ification
    
    4478b739ddade09425d496cbddee1542fdfd5e9b "convert bean to gbuild and add to
    tail_build" broke the following:
    * The name of the officebean dynamic library must not be changed (by adding the
      "lo" extension), as code in officebean.jar uses it in NativeLibraryLoader.
    * With gbuild's per-default hidden visibility, JNIEXPORT functions must instead
      use SAL_DLLPUBLIC_EXPORT.  (Windows-only code in bean/native/win32/ should be
      safe to continue using JNIEXPORT.)
    
    Change-Id: I I3c312dd05c90421ba0c726496b7149a26d155604

diff --git a/Repository.mk b/Repository.mk
index 68a8890..900b538 100644
--- a/Repository.mk
+++ b/Repository.mk
@@ -114,7 +114,6 @@
 	forui \
 	odbc \
 	odbcbase \
-	officebean \
 	oox \
 	ooxml \
 	sdbc \
@@ -276,6 +275,7 @@
 	localedata_euro \
 	localedata_others \
 	mcnttype \
+	officebean \
 	neon \
 	package2 \
     reg \
diff --git a/bean/native/unix/com_sun_star_beans_LocalOfficeWindow.c b/bean/native/unix/com_sun_star_beans_LocalOfficeWindow.c
index a45b49e..cc35283 100644
--- a/bean/native/unix/com_sun_star_beans_LocalOfficeWindow.c
+++ b/bean/native/unix/com_sun_star_beans_LocalOfficeWindow.c
@@ -26,6 +26,8 @@
  *
  ************************************************************************/
 
+#include "sal/config.h"
+
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/Intrinsic.h>
@@ -40,7 +42,7 @@
 #include "jawt_md.h"
 #include "jawt.h"
 
-/*#include "../inc/com_sun_star_comp_beans_LocalOfficeWindow.h"*/
+#include "sal/types.h"
 
 #if defined assert
 #undef assert
@@ -55,7 +57,7 @@
 #define SYSTEM_MAC     5
 #define SYSTEM_XWINDOW 6
 
-JNIEXPORT jlong JNICALL Java_com_sun_star_comp_beans_LocalOfficeWindow_getNativeWindow
+SAL_DLLPUBLIC_EXPORT jlong JNICALL Java_com_sun_star_comp_beans_LocalOfficeWindow_getNativeWindow
   (JNIEnv * env, jobject obj_this);
 
 /*****************************************************************************/
@@ -64,7 +66,7 @@ JNIEXPORT jlong JNICALL Java_com_sun_star_comp_beans_LocalOfficeWindow_getNative
  * Method:    getNativeWindowSystemType
  * Signature: ()I
  */
-JNIEXPORT jint JNICALL Java_com_sun_star_beans_LocalOfficeWindow_getNativeWindowSystemType
+SAL_DLLPUBLIC_EXPORT jint JNICALL Java_com_sun_star_beans_LocalOfficeWindow_getNativeWindowSystemType
   (JNIEnv * env, jobject obj_this)
 {
     (void) env; /* avoid warning about unused parameter */
@@ -79,7 +81,7 @@ JNIEXPORT jint JNICALL Java_com_sun_star_beans_LocalOfficeWindow_getNativeWindow
  * Method:    getNativeWindow
  * Signature: ()J
  */
-JNIEXPORT jlong JNICALL Java_com_sun_star_beans_LocalOfficeWindow_getNativeWindow
+SAL_DLLPUBLIC_EXPORT jlong JNICALL Java_com_sun_star_beans_LocalOfficeWindow_getNativeWindow
   (JNIEnv * env, jobject obj_this)
 {
     (void) env; /* avoid warning about unused parameter */
diff --git a/bean/native/unix/com_sun_star_comp_beans_LocalOfficeWindow.c b/bean/native/unix/com_sun_star_comp_beans_LocalOfficeWindow.c
index 5b36110..8510dc3 100644
--- a/bean/native/unix/com_sun_star_comp_beans_LocalOfficeWindow.c
+++ b/bean/native/unix/com_sun_star_comp_beans_LocalOfficeWindow.c
@@ -26,6 +26,8 @@
  *
  ************************************************************************/
 
+#include "sal/config.h"
+
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/Intrinsic.h>
@@ -40,6 +42,7 @@
 #include "jawt_md.h"
 #include "jawt.h"
 
+#include "sal/types.h"
 
 #define SYSTEM_WIN32   1
 #define SYSTEM_WIN16   2
@@ -71,7 +74,7 @@ static void ThrowException(JNIEnv * env, char const * type, char const * msg) {
  * Method:    getNativeWindowSystemType
  * Signature: ()I
  */
-JNIEXPORT jint JNICALL Java_com_sun_star_comp_beans_LocalOfficeWindow_getNativeWindowSystemType
+SAL_DLLPUBLIC_EXPORT jint JNICALL Java_com_sun_star_comp_beans_LocalOfficeWindow_getNativeWindowSystemType
   (JNIEnv * env, jobject obj_this)
 {
     (void) env; /* avoid warning about unused parameter */
@@ -86,7 +89,7 @@ JNIEXPORT jint JNICALL Java_com_sun_star_comp_beans_LocalOfficeWindow_getNativeW
  * Method:    getNativeWindow
  * Signature: ()J
  */
-JNIEXPORT jlong JNICALL Java_com_sun_star_comp_beans_LocalOfficeWindow_getNativeWindow
+SAL_DLLPUBLIC_EXPORT jlong JNICALL Java_com_sun_star_comp_beans_LocalOfficeWindow_getNativeWindow
   (JNIEnv * env, jobject obj_this)
 {
     jboolean result;
diff --git a/scp2/source/ooo/file_library_ooo.scp b/scp2/source/ooo/file_library_ooo.scp
index 63b2f8f..83f125e 100644
--- a/scp2/source/ooo/file_library_ooo.scp
+++ b/scp2/source/ooo/file_library_ooo.scp
@@ -914,7 +914,16 @@ STD_LIB_FILE( gid_File_Lib_Offacc, offacc)
 #ifdef SOLAR_JAVA
 #ifndef MACOSX
 
-STD_LIB_FILE(gid_File_Lib_Officebean, officebean)
+File gid_File_Lib_Officebean
+    TXT_FILE_BODY;
+    Styles = (PACKED);
+    Dir = SCP2_OOO_BIN_DIR;
+  #ifdef UNX
+    Name = STRING(CONCAT2(libofficebean,UNXSUFFIX));
+  #else
+    Name = "officebean.dll";
+  #endif
+End
 
 #endif
 #endif
commit 6752930497d1c9d0d1ebb29113e89bdaf9e53aaa
Author: Stephan Bergmann <sbergman@redhat.com>
Date:   Mon May 7 16:49:54 2012 +0200

    LocalOfficeConnection description must end in ServiceManager
    
    ...instead of NamingService.  Even though current
    <http://wiki.services.openoffice.org/w/index.php?title=Documentation/DevGuide/JavaBean/Customized_Configuration&oldid=127145>
    documents using the latter, it leads to NullPointerException, as connect()
    expects aInitialObject to be an XPropertySet.
    
    Change-Id: Ia8445bed5307bf440d078a8b96200e3b7fa38706

diff --git a/bean/com/sun/star/comp/beans/LocalOfficeConnection.java b/bean/com/sun/star/comp/beans/LocalOfficeConnection.java
index bb6abdd..4b5f4fc 100644
--- a/bean/com/sun/star/comp/beans/LocalOfficeConnection.java
+++ b/bean/com/sun/star/comp/beans/LocalOfficeConnection.java
@@ -531,7 +531,7 @@ public class LocalOfficeConnection
         throws java.net.MalformedURLException
     {
         // Extruct parameters.
-        int idx = url.indexOf(";urp;StarOffice.NamingService");
+        int idx = url.indexOf(";urp;StarOffice.ServiceManager");
         if (idx < 0)
             throw new java.net.MalformedURLException(
                 "Invalid UNO connection URL.");
